go/libraries/doltcore/table/editor/pk_table_editor.go

```diff
@@ -684,6 +708,45 @@ func (te *pkTableEditor) Table(ctx context.Context) (*doltdb.Table, error) {
 	return te.t, nil
 }
 
+func (te *pkTableEditor) updateAutoIncrementTracker(val interface{}) error {
```

Comment:
This is suspect. Why do we need to manage the state explicitly like this? It should happen automatically as auto inc keys are handed out, no need to tell anyone about it

Reply:
What happens if someone skips ahead on the current reserved value. Something still needs to tell the tracker to update